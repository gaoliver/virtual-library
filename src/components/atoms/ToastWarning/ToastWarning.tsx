import React from 'react';
import {Box, Text, Toast} from 'native-base';

interface ToastWarningProps {
  description: string;
}

const ToastModel: React.FC<ToastWarningProps> = ({description}) => (
  <Box w="100%" p="4" bgColor="warning" borderRadius={10}>
    <Text color="white" fontWeight={'semibold'}>
      {description}
    </Text>
  </Box>
);

export const showToast = (message: string) => {
  return Toast.show({
    duration: 5000,
    render: () => <ToastModel description={message} />,
  });
};
