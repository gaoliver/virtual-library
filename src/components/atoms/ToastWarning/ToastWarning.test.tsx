import React from 'react';
import renderer from 'react-test-renderer';
import {ToastModel, showToast} from './ToastWarning';
import {describe} from '@jest/globals';
import {NativeBaseProvider} from 'native-base';

describe('ToastWarning component UI and function', () => {
  it('ToastModel renders correctly', () => {
    const tree = renderer
      .create(
        <NativeBaseProvider>
          <ToastModel description="Test Description" />
        </NativeBaseProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('showToast renders correctly', () => {
    const tree = renderer.create(showToast('Test Message')).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
