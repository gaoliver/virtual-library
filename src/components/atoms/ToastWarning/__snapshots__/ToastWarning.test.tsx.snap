// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastWarning component UI and function ToastModel renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`ToastWarning component UI and function showToast renders correctly 1`] = `null`;
